assembl/graphql/schema.py

```diff
@@ -387,14 +608,265 @@ def mutate(root, args, context, info):
+        allowed = thematic.user_can(user_id, CrudPermissions.UPDATE, permissions)
+        if not allowed or (allowed == IF_OWNED and user_id == Everyone):
+            raise HTTPUnauthorized()
+
+        with cls.default_db.no_autoflush:
+            title_entries = args.get('title_entries')
+            if len(title_entries) == 0:
+                raise Exception('Thematic titleEntries needs at least one entry')
+                # Better to have this message than
+                # 'NoneType' object has no attribute 'owner_object'
+                # when creating the saobj below if title=None
+
+            update_langstring_from_input_entries(thematic, 'title', title_entries)
+            update_langstring_from_input_entries(thematic, 'description', args.get('description_entries'))
+            kwargs = {}
+            video = args.get('video')
+            if video is not None:
+                update_langstring_from_input_entries(thematic, 'video_title', video['title_entries'])
+                update_langstring_from_input_entries(thematic, 'video_description', video['description_entries'])
+                kwargs['video_html_code'] = video['html_code']
+
+            # take the first entry and set it for short_title
+            kwargs['short_title'] = title_entries[0]['value']
+            kwargs['identifier'] = args.get('identifier')
+            for attr, value in kwargs.items():
+                setattr(thematic, attr, value)
+            db = thematic.db
+            db.flush()
+
+            questions_input = args.get('questions')
+            existing_questions = {question.id: question for question in thematic.get_children()}
+            updated_questions = set()
+            if questions_input is not None:
+                for question_input in questions_input:
+                    if question_input.get('id', None) is not None:
+                        id_ = int(Node.from_global_id(question_input['id'])[1])
+                        updated_questions.add(id_)
+                        question = models.Question.get(id_)
+                        update_langstring_from_input_entries(
+                            question, 'title', question_input['title_entries'])
+                    else:
+                        title_ls = langstring_from_input_entries(
+                            question_input['title_entries'])
+                        models.Question(
+                            title=title_ls,
+                            discussion_id=discussion_id
+                        )
+                        thematic.children.append(
+                            models.Question(
+                                title=title_ls,
+                                discussion_id=discussion_id
+                            )
+                        )
+
+            # remove question (tombstone it) that are not in questions_input
+            for question_id in set(existing_questions.keys()
```

Comment:
I guess. We will use this mutation from the admin interface, we'll send all the json with the thematic and questions.